esp-hal/src/uart.rs

```diff
@@ -2547,6 +2698,32 @@ impl Info {
         txfifo_rst(self.regs(), true);
         txfifo_rst(self.regs(), false);
     }
+
+    fn get_baudrate(&self, clk: u32) -> u32 {
```

Comment:
There are a number of identical lines in each branch here, probably could use some de-duplication.